specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2025-02-01-preview/diagnostic.json

```diff
@@ -1025,21 +1047,53 @@
         "vmCategories": {
           "type": "array",
           "items": {
-            "type": "string"
+            "type": "string",
+            "enum": [
+              "General Purpose",
+              "Compute Optimized",
+              "Memory Optimized",
+              "Storage Optimized",
+              "GPU Accelerated",
+              "FPGA Accelerated",
+              "High Performance Compute",
+              "Confidential Compute"
+            ],
+            "x-ms-enum": {
+              "name": "VmCategories",
+              "modelAsString": false
+            }
           },
           "description": "The vm categories."
         },
         "architectureTypes": {
           "type": "array",
           "items": {
-            "type": "string"
+            "type": "string",
+            "enum": [
+              "x64",
+              "Arm64"
+            ],
+            "x-ms-enum": {
+              "name": "ArchitectureTypes",
+              "modelAsString": false
+            }
           },
           "description": "The cpu architecture types."
         },
         "cpuManufacturers": {
           "type": "array",
           "items": {
-            "type": "string"
+            "type": "string",
+            "enum": [
+              "Intel",
+              "AMD",
+              "Microsoft",
+              "Ampere"
+            ],
+            "x-ms-enum": {
+              "name": "CpuManufacturers",
+              "modelAsString": false
```

Comment:
ditto

Reply:
AS of now no SDK will be generated and as we plan on SDK it will be newer version. So at that point will make this change.